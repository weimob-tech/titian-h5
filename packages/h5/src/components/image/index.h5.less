@import '../common/static/utils.less';
@import '../common/static/var.less';
@import '../common/static/mixins/index.less';

:host {
  display: contents;
}
@image:~ '@{project}-image';

.@{image} {
  display: inline-block;
  position: relative;
  z-index: 0;
  width: 100%;
  overflow: hidden;

  outline: none;

  vertical-align: middle;

  &-ratio::after {
    content: '';

    display: block;
    padding-bottom: calc(1 / (@image-aspect-ratio) * 100%);
  }
  @supports (aspect-ratio: 1) {
    &-ratio {
      aspect-ratio: @image-aspect-ratio;

      &::after {
        content: none;
      }
    }
  }

  &-notice {
    position: absolute;
    z-index: -1;
    top: 50%;
    left: 50%;

    transform: translate(-50%, -50%);

    color: @neutral-color-5;
    font-size: @T28;
    white-space: nowrap;

    --loading-color: @neutral-color-5;
  }

  &-img {
    vertical-align: middle;

    &-ratio {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
    }

    &-fill {
      width: 100%;
      height: 100%;
    }

    &-contain {
      object-fit: contain;

      width: 100%;
      height: 100%;
    }

    &-cover {
      object-fit: cover;

      width: 100%;
      height: 100%;
    }

    &-scaledown {
      object-fit: scale-down;

      width: 100%;
      height: 100%;
    }

    &-center {
      position: absolute;
      top: 50%;
      left: 50%;

      transform: translate(-50%, -50%);
    }

    &-top {
      position: absolute;
      top: 0;
      left: 50%;

      transform: translateX(-50%);
    }

    &-bottom {
      position: absolute;
      bottom: 0;
      left: 50%;

      transform: translateX(-50%);
    }

    &-left {
      position: absolute;
      top: 50%;
      left: 0;

      transform: translateY(-50%);
    }

    &-right {
      position: absolute;
      top: 50%;
      right: 0;

      transform: translateY(-50%);
    }

    &-topleft {
      position: absolute;
      top: 0;
      left: 0;
    }

    &-topright {
      position: absolute;
      top: 0;
      right: 0;
    }

    &-bottomleft {
      position: absolute;
      bottom: 0;
      left: 0;
    }

    &-bottomright {
      position: absolute;
      right: 0;
      bottom: 0;
    }

    &-widthfix {
      width: 100%;
      height: auto;
    }

    &-heightfix {
      width: auto;
      height: 100%;
    }
  }
}
