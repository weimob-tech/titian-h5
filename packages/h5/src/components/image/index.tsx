import { Component, Prop, Element, State, h, Listen, Watch, Event, EventEmitter } from '@stencil/core';
import { JSXBase } from '@stencil/core/internal';
import { BasicComponentAbstract } from '../common/basic/BasicComponent';
import { stringToAttrStyle, throttle, addShadowRootStyle } from '../common/utils';
import * as namespace from '../common/utils/namespace';
import addUnit from '../common/utils/suffix';

enum ImageModeEnum {
  'SCALE_TO_FILL' = 'scaleToFill',
  'ASPECT_FIT' = 'aspectFit',
  'ASPECT_FILL' = 'aspectFill',
  'WIDTH_FIX' = 'widthFix',
  'HEIGHT_FIX' = 'heightFix',
  'TOP' = 'top',
  'BOTTOM' = 'bottom',
  'CENTER' = 'center',
  'LEFT' = 'left',
  'RIGHT' = 'right',
  'TOP_LEFT' = 'top left',
  'TOP_RIGHT' = 'top right',
  'BOTTOM_LEFT' = 'bottom left',
  'BOTTOM_RIGHT' = 'bottom right',
  'SCALE_DOWN' = 'scaleDown',
  'NONE' = 'none',
  'FILL' = 'fill',
  'COVER' = 'cover',
  'CONTAIN' = 'contain',
}
type MainStyleAttrs = {
  'width'?: string;
  'height'?: string;
  'border-radius'?: string;
};
@Component({
  tag: 'ti-image',
  styleUrls: {
    pc: 'index.pc.less',
    h5: 'index.h5.less',
  },
  shadow: true,
})
export class TiImage implements BasicComponentAbstract {
  @Element() host: HTMLElement;

  @Prop() extCss = '';

  @Prop() extClass?: string = '';

  @Prop() extStyle?: string | JSXBase.HTMLAttributes<Record<string, unknown>>['style'];

  @Prop() src: string;

  @Prop() alt: string;

  @Prop() width?: string | number;

  @Prop() height?: string | number;

  @Prop() radius?: string | number;

  @Prop() mode?: ImageModeEnum = ImageModeEnum.SCALE_TO_FILL;

  @Prop() lazyLoad?: boolean;

  @Prop() showLoading?: boolean = true;

  @Prop() showError?: boolean = true;

  @Prop() useLoadingSlot?: boolean;

  @Prop() useErrorSlot?: boolean;

  @Prop() loadingIcon?: string = 'default-pic';

  @Prop() errorIcon?: string = 'default-pic';

  @Prop() loadingIconSize?: number = 72;

  @Prop() errorIconSize?: number = 72;

  @Prop() useGlobalStyle?: boolean;

  @Prop() aspectRatio?: number | 'inherit' = -1;

  @State() loading = true;

  @State() isError = false;

  @State() imageWidth: number;

  @State() imageHeight: number;

  @State() show = false;

  @State() mainStyle: MainStyleAttrs = {};

  @State() imgStyle = {};

  @State() modeClass = {};

  @Event({ bubbles: false, composed: false }) load: EventEmitter<{ width: number; height: number }>;

  @Event({ bubbles: false, composed: false }) error: EventEmitter<never>;

  private handleLazyLoad: any;

  private imgBoxRef = null;

  private imgRef = null;

  private isSupportLoading = false;

  private timer = null;

  @Listen('scroll', { target: 'window', capture: true })
  handleScroll() {
    if (!this.lazyLoad || !this.loading || !this.handleLazyLoad) return;
    this.handleLazyLoad();
  }

  @Watch('radius')
  @Watch('width')
  @Watch('height')
  @Watch('aspectRatio')
  @Watch('extStyle')
  getMainStyle() {
    const { width, height, radius, extStyle, useGlobalStyle, aspectRatio } = this;
    const mainStyle: MainStyleAttrs = {};
    if (useGlobalStyle) {
      const curRadius = radius ? addUnit(radius) : '0px';
      mainStyle['border-radius'] = `calc(var(--base-radius-size, 0px) + ${curRadius})`;
    } else {
      mainStyle['border-radius'] = addUnit(radius);
    }
    if (aspectRatio > 0) {
      mainStyle['--image-aspect-ratio'] = aspectRatio;
    }

    if (width) {
      mainStyle.width = addUnit(width);
    }
    if (height) {
      mainStyle.height = addUnit(height);
      // 没有宽度，只有高度（且不是百分比）
      const supportsAspect = CSS.supports('aspect-ratio: 1');
      const useAspectRatio = aspectRatio > 0 || aspectRatio === 'inherit';
      if (
        !supportsAspect &&
        useAspectRatio &&
        !width &&
        (typeof height === 'string' ? height.indexOf('%') === -1 : true)
      ) {
        mainStyle.width = `calc(${addUnit(height)} * var(--image-aspect-ratio, 1))`;
      }
    }

    this.mainStyle = Object.assign(mainStyle, stringToAttrStyle(extStyle));
  }

  @Watch('mode')
  getImgStyle() {
    const { mode } = this;
    const MODE_MAP = {
      none: 'center',
      scaleToFill: 'fill',
      aspectFill: 'cover',
      aspectFit: 'contain',
    };
    const currentMode = MODE_MAP[mode] || mode || ImageModeEnum.SCALE_TO_FILL;
    this.modeClass = currentMode.toLowerCase().replace(/\s/g, '');
  }

  @Watch('src')
  update(val, oldVal) {
    if (val !== oldVal) {
      this.loading = true;
      this.isError = false;
      this.LazyLoadImg();
    }
  }

  onLoad = () => {
    this.isError = false;
    this.loading = false;
    this.handleLazyLoad = null;
    this.load.emit({ width: this.imgRef.naturalWidth, height: this.imgRef.naturalHeight });
  };

  onError = () => {
    this.isError = true;
    this.loading = false;
    this.error.emit();
  };

  componentWillLoad() {
    this.isSupportLoading = 'loading' in document.createElement('img');
    this.getMainStyle();
    this.getImgStyle();
    addShadowRootStyle.call(this);
  }

  LazyLoadImg() {
    if (!this.lazyLoad || this.isSupportLoading) return;
    const isOutScreen = () => {
      const { top, bottom } = this.imgBoxRef.getBoundingClientRect();
      return top > window.innerHeight * 3 || bottom < window.innerHeight * -3;
    };
    this.handleLazyLoad = throttle(() => {
      if (isOutScreen()) return;
      this.imgRef.src = this.src;
    });
    this.handleLazyLoad();
  }

  onTouchStart = () => {
    const img = this.host.querySelector('img');
    if (img) {
      img.src = this.src;
      return;
    }
    this.timer = setTimeout(() => {
      this.longpress();
    }, 500);
  };

  onTouchMove = () => {
    clearTimeout(this.timer);
  };

  onTouchEnd = () => {
    clearTimeout(this.timer);
  };

  longpress = () => {
    const img = document.createElement('img');
    img.src = this.src;
    img.setAttribute('style', 'opacity: 0;position:absolute;top:0;left:0;width:100%;height:100%');
    this.host.appendChild(img);
  };

  componentDidLoad() {
    this.LazyLoadImg();
  }

  render() {
    const {
      src,
      alt,
      extClass,
      mainStyle,
      imgStyle,
      loading,
      isError,
      modeClass,
      showLoading,
      useLoadingSlot,
      loadingIcon,
      loadingIconSize,
      useErrorSlot,
      showError,
      errorIcon,
      errorIconSize,
      isSupportLoading,
      lazyLoad,
      aspectRatio,
    } = this;
    const useAspectRatio = aspectRatio > 0 || aspectRatio === 'inherit';
    const imgClass = `${namespace.join('image-img', [modeClass, { ratio: useAspectRatio }])}`;
    const iconClass = `${namespace.join('image-notice')}`;
    return (
      <div
        class={`${namespace.join('image', [{ ratio: useAspectRatio }])} ${extClass}`}
        style={mainStyle}
        part={extClass}
        ref={el => el && (this.imgBoxRef = el)}
        aria-hidden="true"
        onTouchStart={this.onTouchStart}
        onTouchMove={this.onTouchMove}
        onTouchEnd={this.onTouchEnd}
      >
        {!isError &&
          (isSupportLoading || !lazyLoad ? (
            <img
              ref={el => el && (this.imgRef = el)}
              class={imgClass}
              src={src}
              alt={alt}
              style={imgStyle}
              loading={lazyLoad ? 'lazy' : 'eager'}
              onLoad={this.onLoad}
              onError={this.onError}
            />
          ) : (
            <img
              ref={el => el && (this.imgRef = el)}
              class={imgClass}
              date-src={src}
              alt={alt}
              style={imgStyle}
              onLoad={this.onLoad}
              onError={this.onError}
            />
          ))}
        {showLoading && loading && (
          <div class={iconClass}>
            {useLoadingSlot ? <slot name="loading" /> : <ti-icon name={loadingIcon} size={loadingIconSize} />}
          </div>
        )}
        {showError && isError && (
          <div class={iconClass}>
            {useErrorSlot ? <slot name="error" /> : <ti-icon name={errorIcon} size={errorIconSize} />}
          </div>
        )}
        <slot />
      </div>
    );
  }
}
