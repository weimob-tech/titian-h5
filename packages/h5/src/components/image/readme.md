# ti-image



<!-- Auto Generated Below -->


## Properties

| Property          | Attribute           | Description      | Type                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Default                       |
| ----------------- | ------------------- | ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------- |
| `alt`             | `alt`               |                  | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | `undefined`                   |
| `errorIcon`       | `error-icon`        |                  | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | `'default-pic'`               |
| `errorIconSize`   | `error-icon-size`   |                  | `number`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | `72`                          |
| `extClass`        | `ext-class`         | 额外的类名，添加到根节点的元素上 | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | `''`                          |
| `extStyle`        | `ext-style`         | 额外的样式            | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | `undefined`                   |
| `height`          | `height`            |                  | `number \| string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | `240`                         |
| `lazyLoad`        | `lazy-load`         |                  | `boolean`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | `undefined`                   |
| `loadingIcon`     | `loading-icon`      |                  | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | `'default-pic'`               |
| `loadingIconSize` | `loading-icon-size` |                  | `number`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | `72`                          |
| `mode`            | `mode`              |                  | `ImageModeEnum.ASPECT_FILL \| ImageModeEnum.ASPECT_FIT \| ImageModeEnum.BOTTOM \| ImageModeEnum.BOTTOM_LEFT \| ImageModeEnum.BOTTOM_RIGHT \| ImageModeEnum.CENTER \| ImageModeEnum.CONTAIN \| ImageModeEnum.COVER \| ImageModeEnum.FILL \| ImageModeEnum.HEIGHT_FIX \| ImageModeEnum.LEFT \| ImageModeEnum.NONE \| ImageModeEnum.RIGHT \| ImageModeEnum.SCALE_DOWN \| ImageModeEnum.SCALE_TO_FILL \| ImageModeEnum.TOP \| ImageModeEnum.TOP_LEFT \| ImageModeEnum.TOP_RIGHT \| ImageModeEnum.WIDTH_FIX` | `ImageModeEnum.SCALE_TO_FILL` |
| `radius`          | `radius`            |                  | `number \| string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | `undefined`                   |
| `showError`       | `show-error`        |                  | `boolean`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | `true`                        |
| `showLoading`     | `show-loading`      |                  | `boolean`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | `true`                        |
| `src`             | `src`               |                  | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | `undefined`                   |
| `useErrorSlot`    | `use-error-slot`    |                  | `boolean`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | `undefined`                   |
| `useLoadingSlot`  | `use-loading-slot`  |                  | `boolean`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | `undefined`                   |
| `width`           | `width`             |                  | `number \| string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | `320`                         |


## Events

| Event   | Description | Type                                              |
| ------- | ----------- | ------------------------------------------------- |
| `click` |             | `CustomEvent<never>`                              |
| `error` |             | `CustomEvent<never>`                              |
| `load`  |             | `CustomEvent<{ width: number; height: number; }>` |


## Dependencies

### Depends on

- [ti-icon](../icon)

### Graph
```mermaid
graph TD;
  ti-image --> ti-icon
  style ti-image fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
