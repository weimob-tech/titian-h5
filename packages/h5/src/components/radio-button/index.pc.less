@import '../common/static/var';
@import '../common/static/utils';

@radioButton: ~'@{project}-radio-button';

:host {
  display: contents;
}

.@{radioButton} {
  display: inline-flex;
  padding: @radio-button-padding-v @radio-button-padding-h;
  overflow: hidden;

  --tag-background-color: @radio-button-bg-color;
  --tag-text-color: @radio-button-text-color;
  --tag-padding: @radio-button-padding;
  --tag-font-size: @T26;
  --tag-height: @radio-button-height;
  --tag-radius: @radio-button-radius;
  --tag-border-width: 2px;
  --tag-border-color: transparent;

  &-checked {
    --tag-border-color: @radio-button-checked-border-color;
    --tag-background-color: @radio-button-checked-bg-color;
    --tag-text-color: @radio-button-checked-text-color;
  }

  &-disabled {
    pointer-events: none;

    --tag-text-color: rgba(117, 117, 117, 40%);
  }

  &-disabled.@{radioButton}-checked {
    --tag-text-color: @radio-button-disabled-text-color;
    --tag-border-color: @radio-button-disabled-border-color;
  }
}
