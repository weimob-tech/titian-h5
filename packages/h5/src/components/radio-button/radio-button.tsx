import { Component, Host, h, Prop, Method, EventEmitter, Element, State, Event, Watch } from '@stencil/core';
import { JSXBase } from '@stencil/core/internal';
import { Components } from '../../components';
import { BasicComponentAbstract } from '../common/basic/BasicComponent';
import { addShadowRootStyle, stringToAttrStyle } from '../common/utils';
import { hexToRGB, RGBToRGBA } from '../common/utils/color';
import * as namespace from '../common/utils/namespace';
import { getParent } from '../common/utils/relation';

type ParentAttrs = Components.TiRadioGroup & {
  privateDefaultValue?: string | number;
  change?: EventEmitter<string | number>;
};

@Component({
  tag: 'ti-radio-button',
  styleUrls: {
    pc: 'index.pc.less',
    h5: 'index.h5.less',
  },
  shadow: true,
})
export class TiRadioButton implements BasicComponentAbstract {
  @Element() host: HTMLElement;

  @Prop() extClass = '';

  @Prop() extStyle?: string | JSXBase.HTMLAttributes<Record<string, unknown>>['style'];

  @Prop() label?: string;

  @Prop() value?: string | number = null;

  @Prop() disabled?: boolean | null = null;

  @Prop() checked?: boolean = null;

  @Prop() defaultChecked?: boolean = null;

  @Prop() color?: string;

  @Prop() icon = '';

  @State() privateChecked = false;

  @Event({ bubbles: true, composed: false }) change: EventEmitter<boolean>;

  @State() toggleClass = '';

  @Method()
  async updateDataFromParent(parent: ParentAttrs) {
    if (parent) {
      const parentInstance = await parent?.getInstance();
      const { disabled, value: parentValue, privateDefaultValue, icon, color } = parentInstance || parent;
      const { value } = this;

      let privateChecked = false;
      let checked = null;
      let defaultChecked = null;

      if (parentValue !== null) {
        checked = parentValue === value;
      }

      if (privateDefaultValue !== null) {
        defaultChecked = privateDefaultValue === value;
      }

      if (defaultChecked !== null) {
        privateChecked = defaultChecked;
      }

      if (checked !== null) {
        privateChecked = checked;
      }

      this.checked = checked;
      this.icon = icon ?? this.icon;
      this.color = color ?? this.color;
      this.defaultChecked = defaultChecked;
      this.privateChecked = privateChecked;
      this.disabled = disabled ?? this.disabled;
    }
  }

  get parent() {
    return getParent({
      host: this.host,
      tag: 'ti-radio-group',
      relations: 'ancestor',
    })?.[0];
  }

  @Watch('checked')
  watchChecked(value: boolean, oldValue: boolean) {
    if (value !== oldValue && !this.parent) {
      if (value !== undefined && this.privateChecked !== value) {
        this.privateChecked = value;
      }
    }
  }

  componentWillLoad(): void | Promise<void> {
    let checked = false;
    if (this.defaultChecked !== null) {
      checked = this.defaultChecked;
    }

    if (this.checked !== null) {
      checked = this.checked;
    }
    this.privateChecked = checked;

    addShadowRootStyle.call(this);

    this.updateDataFromParent(this.parent);
  }

  private handleChange() {
    const { privateChecked, checked, value, parent: parentEl } = this;
    if (parentEl) {
      parentEl.getInstance().then((parent: ParentAttrs) => {
        parent.change.emit(value);
        parent.privateDefaultValue = value;
      });
      return;
    }
    const nextChecked = !privateChecked;
    this.change.emit(nextChecked);
    if (checked === null) {
      this.privateChecked = nextChecked;
    }
  }

  private handleTagClick() {
    const { disabled } = this;
    if (disabled) return;
    this.handleChange();
  }

  private getClassNames() {
    const { disabled, privateChecked, extClass } = this;
    const classList = [namespace.join('radio-button', [{ disabled, checked: privateChecked }]), extClass];

    return classList.join(' ');
  }

  private formatColor(colorStr, opacity = 1) {
    if (!colorStr) return '';
    let rgba = '';
    if (colorStr.includes('#')) {
      rgba = hexToRGB(colorStr, opacity);
    }
    if (colorStr.includes('rgba')) {
      const reg = /^rgba?\((.+)\)$/g;
      const a = colorStr.replace(reg, (_s, $1) => $1);
      const [r, g, b] = a.split(',');
      rgba = `rgba(${r}, ${g}, ${b}, ${opacity})`;
    }
    if (colorStr.includes('rgb')) {
      rgba = RGBToRGBA(colorStr, opacity);
    }

    return rgba;
  }

  render() {
    const { label, icon, color } = this;
    const styles: Record<string, string> = color
      ? {
          '--radio-button-disabled-border-color': this.formatColor(this.color, 0.2),
          '--radio-button-disabled-text-color': this.formatColor(this.color, 0.4),
          '--radio-button-checked-bg-color': this.formatColor(this.color, 0.1),
          '--radio-button-checked-text-color': this.formatColor(this.color, 1),
          '--radio-button-checked-border-color': this.formatColor(this.color, 0.4),
        }
      : {};
    return (
      <Host>
        <div
          part={this.extClass}
          class={this.getClassNames()}
          style={{ ...styles, ...stringToAttrStyle(this.extStyle) }}
        >
          <ti-tag onClick={this.handleTagClick.bind(this)} rightIcon={icon}>
            <slot name="prefix" slot="prefix" />
            {label || <slot />}
            <slot name="suffix" slot="suffix" />
          </ti-tag>
        </div>
      </Host>
    );
  }
}
