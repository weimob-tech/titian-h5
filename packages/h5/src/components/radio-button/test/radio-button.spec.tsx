import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { TestContainer } from '../../common/test/container';
import { TiImage } from '../../image';
import { TiRadioButton } from '../radio-button';

describe('渲染单选按钮', () => {
  it('基本渲染', async () => {
    const page = await newSpecPage({
      components: [TiRadioButton, TestContainer],
      template: () => (
        <test-container>
          <ti-radio-button />
          <ti-radio-button label="文字内容" />
          <ti-radio-button>文字内容</ti-radio-button>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('默认选中状态', async () => {
    const page = await newSpecPage({
      components: [TiRadioButton, TestContainer],
      template: () => (
        <test-container>
          <ti-radio-button label="默认选中" checked />
          <ti-radio-button label="默认选中" default-checked />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
  //

  it('组件内容设置', async () => {
    const page = await newSpecPage({
      components: [TiRadioButton, TiImage, TestContainer],
      template: () => (
        <test-container>
          <ti-radio-button label="设置右侧图标" icon="sort-inactive" />
          <ti-radio-button label="自定义左侧内容">
            <ti-image
              ext-style="margin-right: 6px"
              slot="prefix"
              width="40"
              radius="0"
              height="40"
              src="https://image-c-dev.weimobwmc.com/qa-On6X/8b97cd488593474ba4a8ccaa3c1a493f.png"
            />
          </ti-radio-button>
          <ti-radio-button label="自定义右侧内容">
            <ti-image
              ext-style="margin-left: 6px"
              slot="suffix"
              width="40"
              radius="0"
              height="40"
              src="https://image-c-dev.weimobwmc.com/qa-On6X/8b97cd488593474ba4a8ccaa3c1a493f.png"
            />
          </ti-radio-button>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('改变单选按钮颜色', async () => {
    const page = await newSpecPage({
      components: [TiRadioButton, TestContainer],
      template: () => (
        <test-container>
          <ti-radio-button label="通过属性更改颜色" color="#2a6ae9" />
          <ti-radio-button label="通过属性更改颜色" color="rgb(0,0,0)" />
          <ti-radio-button label="通过属性更改颜色" color="rgba(0,0,0, 0.5)" />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('受控模式', async () => {
    const change = jest.fn();
    const page = await newSpecPage({
      components: [TiRadioButton],
      template: () => <ti-radio-button label="监听切换状态变化" checked onChange={change} />,
    });
    {
      // 点击
      const tag = page.root.shadowRoot.querySelector<HTMLDivElement>('ti-tag');
      tag.click();

      await page.waitForChanges();
      expect(change.mock.lastCall[0].detail).toBeFalsy();

      tag.click();
      await page.waitForChanges();
      expect(change.mock.lastCall[0].detail).toBeFalsy();

      page.root.setAttribute('checked', 'false');
      await page.waitForChanges();
      tag.click();
      await page.waitForChanges();
      expect(change.mock.lastCall[0].detail).toBeTruthy();
    }
  });

  it('监听点击切换事件', async () => {
    const change = jest.fn();
    const page = await newSpecPage({
      components: [TiRadioButton],
      template: () => <ti-radio-button label="监听切换状态变化" onChange={change} />,
    });
    {
      // 点击
      const tag = page.root.shadowRoot.querySelector<HTMLDivElement>('ti-tag');
      tag.click();

      await page.waitForChanges();
      expect(change.mock.lastCall[0].detail).toBeTruthy();
      const radioButton = page.root.shadowRoot.querySelector('.titian-radio-button');
      expect(radioButton).toHaveClass('titian-radio-button-checked');

      tag.click();
      await page.waitForChanges();
      expect(change.mock.lastCall[0].detail).toBeFalsy();
      expect(radioButton).not.toHaveClass('titian-radio-button-checked');
    }
  });
});
