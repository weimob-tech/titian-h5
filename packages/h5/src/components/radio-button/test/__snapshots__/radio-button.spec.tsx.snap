// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染单选按钮 基本渲染 1`] = `
<test-container>
  <!---->
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button" part="">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          <slot></slot>
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
  </ti-radio-button>
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button" part="">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          文字内容
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
  </ti-radio-button>
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button" part="">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          <slot></slot>
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
    文字内容
  </ti-radio-button>
</test-container>
`;

exports[`渲染单选按钮 改变单选按钮颜色 1`] = `
<test-container>
  <!---->
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button" part="" style="--radio-button-disabled-border-color: rgba(42, 106, 233, 0.2); --radio-button-disabled-text-color: rgba(42, 106, 233, 0.4); --radio-button-checked-bg-color: rgba(42, 106, 233, 0.1); --radio-button-checked-text-color: rgba(42, 106, 233, 1); --radio-button-checked-border-color: rgba(42, 106, 233, 0.4);">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          通过属性更改颜色
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
  </ti-radio-button>
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button" part="" style="--radio-button-disabled-border-color: rgba(0,0,0,0.2); --radio-button-disabled-text-color: rgba(0,0,0,0.4); --radio-button-checked-bg-color: rgba(0,0,0,0.1); --radio-button-checked-text-color: rgba(0,0,0,1); --radio-button-checked-border-color: rgba(0,0,0,0.4);">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          通过属性更改颜色
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
  </ti-radio-button>
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button" part="" style="--radio-button-disabled-border-color: rgba(0,0,0,0.2); --radio-button-disabled-text-color: rgba(0,0,0,0.4); --radio-button-checked-bg-color: rgba(0,0,0,0.1); --radio-button-checked-text-color: rgba(0,0,0,1); --radio-button-checked-border-color: rgba(0,0,0,0.4);">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          通过属性更改颜色
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
  </ti-radio-button>
</test-container>
`;

exports[`渲染单选按钮 组件内容设置 1`] = `
<test-container>
  <!---->
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button" part="">
        <ti-tag righticon="sort-inactive">
          <slot name="prefix" slot="prefix"></slot>
          设置右侧图标
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
  </ti-radio-button>
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button" part="">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          自定义左侧内容
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
    <ti-image ext-style="margin-right: 6px" slot="prefix">
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-image" part="" style="border-radius: 0px; width: 20px; height: 20px; margin-right:  6px;">
          <img class="titian-image-img titian-image-img-fill" loading="eager" src="https://image-c-dev.weimobwmc.com/qa-On6X/8b97cd488593474ba4a8ccaa3c1a493f.png">
          <div class="titian-image-notice">
            <ti-icon name="default-pic" size="72"></ti-icon>
          </div>
          <slot></slot>
        </div>
      </mock:shadow-root>
    </ti-image>
  </ti-radio-button>
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button" part="">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          自定义右侧内容
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
    <ti-image ext-style="margin-left: 6px" slot="suffix">
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-image" part="" style="border-radius: 0px; width: 20px; height: 20px; margin-left:  6px;">
          <img class="titian-image-img titian-image-img-fill" loading="eager" src="https://image-c-dev.weimobwmc.com/qa-On6X/8b97cd488593474ba4a8ccaa3c1a493f.png">
          <div class="titian-image-notice">
            <ti-icon name="default-pic" size="72"></ti-icon>
          </div>
          <slot></slot>
        </div>
      </mock:shadow-root>
    </ti-image>
  </ti-radio-button>
</test-container>
`;

exports[`渲染单选按钮 默认选中状态 1`] = `
<test-container>
  <!---->
  <ti-radio-button>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button titian-radio-button-checked" part="">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          默认选中
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
  </ti-radio-button>
  <ti-radio-button default-checked="">
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-radio-button titian-radio-button-checked" part="">
        <ti-tag righticon="">
          <slot name="prefix" slot="prefix"></slot>
          默认选中
          <slot name="suffix" slot="suffix"></slot>
        </ti-tag>
      </div>
    </mock:shadow-root>
  </ti-radio-button>
</test-container>
`;
